### PR TITLE
Plane: set_servos_controlled rework

### PR DESCRIPTION
**Pull request by IamPete1**
This is another minor logical rework to the `set_servos_controlled` function.  No change in behavior.

### CODE REVIEW

File: ArduPlane/mode_manual.cpp [hunk 1]
```diff
@@ -6,7 +6,7 @@ void ModeManual::update()
     SRV_Channels::set_output_scaled(SRV_Channel::k_aileron, plane.roll_in_expo(false));
     SRV_Channels::set_output_scaled(SRV_Channel::k_elevator, plane.pitch_in_expo(false));
     output_rudder_and_steering(plane.rudder_in_expo(false));
-    float throttle = plane.get_throttle_input(true);
+    float throttle = rc().has_valid_input() ? plane.get_throttle_input(true) : 0.0;
```
Comment (tridge) [inline ArduPlane/mode_manual.cpp:9]: in plane we handle this with default values in radio.cpp, see control_failsafe(). This seems to be defeating that?

**Review by tridge — CHANGES_REQUESTED** [review]
this doesn't look right to me - the plane strategy is to replace the rc input values with some defaults when in failsafe, not do it in each use location

**Comment by IamPete1** [comment]
@tridge The rc-input replace only happens once the failsafe has triggered, there is some time between the RC going invalid and the failsafe trigger, or the user can disable the failsafe all together, in which case you get the invalid RC output to the throttle. 

The point here is not really to change the failsafe behavior but to make it consistent between modes. We already do the check here for stabilize/training/acro/FBWA

https://github.com/ArduPilot/ardupilot/blob/37109e649e98d6628d263024155428f91c19af40/ArduPlane/servos.cpp#L550-L551


This patch just makes manual, throttle suppressed pass through in any mode and guided throttle pass through behave in the same way. It also changes quadplane esc calibration and motor test.   

The key change is manual, users are unlikely to notice the others.

**Comment by tridge** [comment]
@IamPete1 for manual mode, I'd like to understand the scenario more. Jumping to zero for the short period between RC failing and RC failsafe kicking in isn't necessary a good thing on some aircraft. For example on some IC engines it will stop the engine.

**Comment by tridge** [comment]
we might be better off freezing throttle for all manual throttle modes for that intermediate period?

**Comment by IamPete1** [comment]
It might be easier to get the `control_failsafe` method working. As I say my main goal is not to change the behavior, I just want that be behavior to be constant. So I had a closer look.

`control_failsafe` uses `rc_failsafe_active` before overriding stabilize/training/acro/FBWA uses `has_valid_input`

https://github.com/ArduPilot/ardupilot/blob/66b8fd95adeb2b5eb05e4fa32a35795b51d5a650/ArduPlane/RC_Channel.cpp#L24-L33

`has_valid_input` calls `in_rc_failsafe` on 26 which checks `rc_failsafe_active` as used in `control_failsafe` 

https://github.com/ArduPilot/ardupilot/blob/66b8fd95adeb2b5eb05e4fa32a35795b51d5a650/ArduPlane/RC_Channel.cpp#L19-L22

So `has_valid_input` could be expanded to:
```
 bool RC_Channels_Plane::has_valid_input() const 
 { 
     if (plane.rc_failsafe_active()) { 
         return false; 
     } 
     if (plane.failsafe.rc_failsafe) { 
         return false; 
     } 
     if (plane.failsafe.throttle_counter != 0) { 
         return false; 
     } 
     return true; 
 } 
```

So `has_valid_input` add two cases where it can return false. `plane.failsafe.rc_failsafe` and `plane.failsafe.throttle_counter` 

Checking `plane.failsafe.throttle_counter` adds some hysteresis because the counter must count down from 10. `read_radio` is called at 50Hz, so 0.2 seconds after `rc_failsafe_active` goes false `plane.failsafe.throttle_counter` will hit 0. 

Like wise `plane.failsafe.rc_failsafe` is set when the counter hits 10 and clears when it hits 0. So in this case I think checking `plane.failsafe.rc_failsafe` and `plane.failsafe.throttle_counter` is redundant.

So, from my review `has_valid_input` is slightly more conservative because of that small delay in clearing. (I think the delay is a good idea, stops us using a signal that is rapidly coming in and out of failsafe) 

Depending on if we like the delay or not we could change `control_failsafe` to use `has_valid_input` or remove the existing checking in the stabilize/training/acro/FBWA and use the existing override from `control_failsafe`.

I guess this PR in its current form is a watered down version of switching `control_failsafe` over to `has_valid_input` because it only affects manual throttle in a few cases. If we were to update `control_failsafe` it affects all modes and roll/pitch/yaw as well as throttle.

**Review by tridge — APPROVED** [review]

**Pull request merged**